.releaserc.js

```diff
@@ -1,34 +1,61 @@
 const { execSync } = require('child_process')
+const micromatch = require('micromatch')
 
-const useLocalBranch = Boolean(process.env.LOCAL_SEMANTIC_RELEASE_BRANCH)
+const localBranchAsMain = Boolean(
+  process.env.LOCAL_SEMANTIC_RELEASE_BRANCH_AS_MAIN,
+)
 const getCurrentBranch = () =>
   execSync('git rev-parse --abbrev-ref HEAD', {
     encoding: 'utf-8',
   }).trim()
 const repositoryUrl = process.env.LOCAL_SEMANTIC_RELEASE_REPOSITORY_URL?.trim()
 const isDotRepositoryUrl = repositoryUrl === '.'
+const MAINTENANCE_BRANCH_GLOB = '*[0-9]+.x'
+const getMaintenanceBranchConfig = () => {
+  const baseConfig = {
+    name: MAINTENANCE_BRANCH_GLOB,
+  }
+  const currentBranch = getCurrentBranch()
+  const isMaintenanceBranch = micromatch.isMatch(
+    currentBranch,
+    MAINTENANCE_BRANCH_GLOB,
+  )
+  if (!isMaintenanceBranch) return baseConfig
+  const prefixedMajorVersion = currentBranch.split('.').at(-2)
+  const matches = /[0-9]+$/.exec(prefixedMajorVersion)
+  const majorVersion = matches[0]
+  if (
+    matches.index === 0 &&
+    prefixedMajorVersion.length === majorVersion.length
+  )
+    throw new Error('Non major version maintenance branches are not supported')
+  return {
+    ...baseConfig,
+    range: `${majorVersion}.x`,
+    channel: `v${majorVersion}`,
+  }
+}
 
+// noinspection JSUnusedGlobalSymbols
 /**
  * @type {import('semantic-release').GlobalConfig}
  */
 module.exports = {
   repositoryUrl,
   branches: [
-    //👇 Fake branch so that we can release beta versions in `main`
-    //   until we can release 1.0.0
-    'semantic-release',
+    //👇 Major version maintenance branches
+    getMaintenanceBranchConfig(),
     {
-      name: useLocalBranch ? getCurrentBranch() : 'main',
-      prerelease: 'beta',
```

Comment:
This was a mistake! See https://github.com/davidlj95/ngx/pull/1146#issuecomment-2827402858

Mistake reason: porting it from branch #1144